core/src/main/java/com/nerds/gamejam/gameplay/planet/PlanetFactory.java

```diff
@@ -37,14 +38,27 @@ public void createPlanet(World world, int x, int y) {
 
         float planetScale = GameJam.randomSeed.getRandomGenerator().nextFloat() + 0.5f;
         List<TextureReference> layers = createTextureReferences(baseMaterial, secondaryMaterial, landmass);
+        BodyComponent body = new BodyComponent(24, 24);
+        float radius = body.width / 2 * planetScale;
+        body.physicalBody = new Circle(x + radius, y + radius, radius);
 
         Entity worldEntity = world.createEntity();
+        String planetName = nameFactory.generatePlanetName();
+
+        Material finalBaseMaterial = baseMaterial;
+        Material finalSecondaryMaterial = secondaryMaterial;
+        Landmass finalLandmass = landmass;
         worldEntity.edit()
                 .add(new PositionComponent(x, y))
-                .add(new BodyComponent(24, 24))
+                .add(body)
                 .add(new TextureReferenceComponent(layers))
                 .add(new ScaleComponent(planetScale, planetScale))
-                .add(new FontComponent(nameFactory.generatePlanetName(), x - 10, y - 10));
+                .add(new FontComponent(planetName, x - 10, y - 10))
+                .add(new ClickableComponent((worldX, worldY, screenX, screenY, button) -> {
+                    String desc = "A " + finalBaseMaterial.getDisplayName() + " planet, covered in " + finalSecondaryMaterial.getDisplayName() + " " + finalLandmass.getDisplayName() + " formations";
```

Comment:
You can put this in the translations too - I think you can do nice placeholdery stuff